### DISCUSSION TITLE
Support sending sim data to MQTT

### DISCUSSION THREAD

**Discussion by neilenns** [MobiFlight/MobiFlight-Connector#1331] [Ideas]
**Is your feature request related to a problem? Please describe.**
It'd be so cool to take data from the sim and send it to an MQTT server for automation purposes.

Simple example: get the time of day from the sim and use it to control the brightness of your gaming room lights to match the actual outdoor brightness.

**Describe the solution you'd like**
MQTT as an output type

**Additional context**

Super basic proof of concept works. This would follow the same pattern as [WebSockets](https://github.com/MobiFlight/MobiFlight-Connector/pull/916) and should be done after that work is merged.

**Comment by neilenns** [comment]
I have a branch with this working for outputs. It's pretty cool.

**Comment by sledzik1984** [comment]
Hi! Is this idea implemented?
I am looking for way to extract data of cockpit temperature from Tollis, then send it to Homeassistant to set target temperature in a room with my Home cockpit :)

Also - this could be used to use cockpit door switch to block door to home cockpit room using "Lock cockpit door" in SIM

**Reply by neilenns** [reply]
Alas no :(

**Comment by tigert** [comment]
I looked at ESPHome again, and they support a ton of different things. Like shift registers and i2c etc.

If you just ewant your cockpit door switch state sent from MobiFlight to HomeAssistant, that would be one way to do it. Set a LED output from MF, and read it as a button sensor into ESPHome. Yes, it would be silly because it is two pieces of software using a hardware bridge to talk to each other, and it would be way more sensible to do it with proper integration, but it would work.

**Comment by tigert** [comment]
@neilenns I have been thinking of this, and I think we should rather do a proper API integration. So that we could discover HomeAssistant on your LAN, query the devices and represent the suitable entities as input and output devices and perhaps grouped by the "area" (room etc) as separate controllers for example. So you could read something like _Home Assistant > Hallway > Doorbell ringing state_ and send it as cabin call to your cockpit. Or adjust your ceiling fan based on your simulator air conditioning blower. I mean nowadays people have a lot more of these things, and they are not all geeks that know how to write yaml and scripts, and those devices are becoming more useful also as simulator setup. Like you could use actual smart lights as your cockpit flood lights instead of wiring your own power supply and leds and connect them to an arduino with a mosfet.

I dont think many users really know how to do that kind of stuff via MQTT even though it would be totally possible, but to make this something actually useful, it should probably work similarly to the ProSim integration for example.

And HomeAssistant is open source and really nice piece of software, so we could do this properly, rather than expect people to mess with mqtt calls.

**Reply by tigert** [reply]
https://github.com/home-assistant/home-assistant-js-websocket

**Reply by tigert** [reply]
That lib is pretty sweet.

I could load my actual weight from my Withings scale and put it into the XCub pilots seat.

**Reply by tigert** [reply]
Also, "scan for input" -> ring your doorbell. Or "listen to HomeAssistant" (I assume there is a way to monitor the HA log feed via the API?) to pick up a device you want to turn on or off from MF.

**Reply by tigert** [reply]
HA helpers could work as virtual input buttons for MF, so you could for example create a simple instructor station as a HA dashboard.

**Reply by tigert** [reply]
Ok the javascript lib is neat, did some ai-assisted poking at it. 

I have "scan for input" and "scan for output" working via HomeAssistant event log, so I can turn a light on and it shows up and I could then assign this in MF. And if I press a zigbee switch, it also shows up, and could then act as a MF input device if we'd implement this in MF side. Right now it is just a proof of concept demo.

This is how it should really work. Hmm.

**Reply by neilenns** [reply]
HA direct access is cool, but I think is a separate request from MQTT. They both let you talk to HA, but MQTT Is a broader (admittedly nerdier) solution that enables any MQTT-enabled system to integrate with MobiFlight.

We should do both ;)
